R/assemble.R

```diff
@@ -162,27 +208,34 @@ parse_survey <- function(surv_obj, oauth_token = getOption('sm_oauth_token'), ..
 
   # reset to text names instead of numbers
   # and then re-order to correct columns
-  names(out)[(length(static_vars) + 1):length(names(out))] <- qid_text_crosswalk$unique_text[match(names(out)[(length(static_vars) + 1):length(names(out))],qid_text_crosswalk$q_unique_id)]
+  names(out)[(length(static_vars) + 1):length(names(out))] <- qid_text_crosswalk$unique_text[match(
+    names(out)[(length(static_vars) + 1):length(names(out))],
+    qid_text_crosswalk$q_unique_id
+  )]
   out <- out[, col_names]
   out <- out %>%
-    dplyr::arrange(dplyr::desc(response_id)) %>%
-    dplyr::rename(respondent_id = response_id)
+    dplyr::arrange(dplyr::desc(.data$response_id)) %>%
+    dplyr::rename(respondent_id = .data$response_id)
 
-   # Join response status
+  # Join response status
   out <- out %>%
-    dplyr::left_join(.,status, by = c("respondent_id" = "id")) %>%
-    dplyr::select(survey_id, collector_id, respondent_id, date_created, date_modified, response_status, everything())
+    dplyr::left_join(., status, by = c("respondent_id" = "id")) %>%
+    dplyr::select(
+      .data$survey_id, .data$collector_id, .data$respondent_id,
+      .data$date_created, .data$date_modified, .data$response_status,
+      tidyselect::everything()
+    )
   out
 }
 
 # Helper function for de-duplicating identical Q names
 # Input: the vector of names
 # Adapted from janitor::make_clean_names()
-de_duplicate_names <- function(x){
+de_duplicate_names <- function(x) {
   dupe_count <- vapply(seq_along(x), function(i) {
     sum(x[i] == x[1:i])
   }, integer(1))
   x[dupe_count > 1] <- paste(x[dupe_count >
-                                 1], dupe_count[dupe_count > 1], sep = "_")
+    1], dupe_count[dupe_count > 1], sep = "_")
```

Comment:
for something that's optimizing for readability, this change seemed strange to me, putting the `1]` on its own line rather than breaking at the comma

Reply:
@sfirke in fairness, that isn't the lintr that is modifying the code. I used the `{styler}` package to modify it on the first pass automatically as there were a lot of lint issues. the linter only checks and reports, it doesn't modify. So you could change this line break to something more sensible and still pass the lint check.

for what it is worth, my natural line break for this is the calling of the function

```
x[dupe_count > 1] <- paste0(
  x[dupe_count > 1], dupe_count[dupe_count > 1],
  sep = "_"
)
```

which should also be accepted by the linter, I can modify this

Reply:
I have now pushed a change to address this